pkgs/by-name/lu/lux-cli/package.nix

```
@@ -0,0 +1,116 @@
+{
+  stdenv,
+  lib,
+  fetchFromGitHub,
+  rustPlatform,
+  pkg-config,
+  openssl,
+  libgit2,
+  gnupg,
+  libgpg-error,
+  gpgme,
+  darwin,
+  installShellFiles,
+  luajit,
+  makeWrapper,
+  symlinkJoin,
+  lua51Packages,
+  lua52Packages,
+  lua53Packages,
+  lua54Packages,
+  nix,
+  versionCheckHook,
+}:
+let
+  lux-lua-bundle = symlinkJoin {
+    name = "lux-lua-bundle";
+    paths = [
+      lua51Packages.lux-lua
+      lua52Packages.lux-lua
+      lua53Packages.lux-lua
+      lua54Packages.lux-lua
+    ];
+  };
+in
+rustPlatform.buildRustPackage rec {
+  pname = "lux-cli";
+
+  version = "0.3.13";
+
+  src = fetchFromGitHub {
+    owner = "nvim-neorocks";
+    repo = "lux";
+    rev = "v${version}";
+    hash = "sha256-G/j6Tz6mZwdpo7QmVF7Vw5BrEhTtKtzoLh0qkWXxjsQ=";
+  };
+
+  buildAndTestSubdir = "lux-cli";
+  useFetchCargoVendor = true;
+  cargoHash = "sha256-duyaLqwiprrGATTA9ZmvzNaOhnrq1EdlCNobBUCFYUc=";
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+  versionCheckProgramArg = "--version";
```

```suggestion
  versionCheckProgram = "${placeholder "out"}/bin/${meta.mainProgram}";
  versionCheckProgramArg = "--version";
```
Looks like default is to try `pname` so we need to tell it to use `mainProgram`

---

I'm confused by this one too @GaetanLepage.
Previously (in other PRs), people have asked me to use `meta.mainProgram` or `lib.getExe`.

---

Searching in nixpkgs I see this design used by other packages.

---

I think the main reason for using `meta.mainProgram` is that it propagates when a user overrides the package.

That might not be so important for a passthru test, but I think it matters here? 
version check attributes are new to me.

---

I would argue that `pname` and `src.repo` are semantically different, but that's not the case here.